go/tools/gopackagesdriver/archive_to_json.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type scanner interface {
```

Comment:
If you can squash this with `//go/tools/builder`, then you can reuse `expandParamsFiles`, which already does this processing for most actions.

If not, this can be simplified quite a bit: it's fine to expand to a big slice of arguments rather than buffering and parsing incrementally.